### DISCUSSION TITLE
Oracle sql outer join syntax

### DISCUSSION THREAD

**Discussion by haarse** [IvorySQL/IvorySQL#681] [General]
Will you fix so that Oracle outer join syntax will work?
Or does it already work? (it does not work for me)

Example:
SELECT o.orderno, c.customername
FROM order o, customer c
WHERE o.customerid = c.customerid (+);

It should become:
SELECT o.orderno, c.customername
LEFT OUTER JOIN o.customerid = c.customerid (+);

This would make a huge inpact for old system transforming to IvorySql from Oracle db.

**Comment by balinorLiang** [comment]
> Will you fix so that Oracle outer join syntax will work? Or does it already work? (it does not work for me)
> 
> Example: SELECT o.orderno, c.customername FROM order o, customer c WHERE o.customerid = c.customerid (+);
> 
> It should become: SELECT o.orderno, c.customername LEFT OUTER JOIN o.customerid = c.customerid (+);
> 
Does this SQL work in Oracle ? I think it needs to be converted to `SELECT o.orderno, c.customername
FROM order o LEFT OUTER JOIN customer c on o.customerid = c.customerid (+);`
> This would make a huge inpact for old system transforming to IvorySql from Oracle db.

**Comment by haarse** [comment]
Sorry I missed to create a SQL that works:
SELECT o.orderno, c.customername
FROM order o
LEFT OUTER JOIN customer c
ON o.customerid = c.customerid;

**Comment by haarse** [comment]
Name the table orders instead of order, this will make this test easier :-)

the full example:
drop table orders;
create table orders (orderno number, customerid number);
drop table customer;
create table customer (customerid number, customername varchar2(100));

insert into orders values (1, 1);
insert into orders values (2,2);
insert into customer values (2, 'Test');

SELECT o.orderno, c.customername
FROM orders o, customer c
WHERE o.customerid = c.customerid(+);

SELECT o.orderno, c.customername
FROM orders o
LEFT OUTER JOIN customer c
ON o.customerid = c.customerid;

**Reply by balinorLiang** [reply]
![image](https://github.com/user-attachments/assets/c2d6798d-b951-4aa5-8739-64ecd72bb251)
I didn't get any syntax error in IvorySQL. And result is same with Oracle.
![image](https://github.com/user-attachments/assets/8af8ec0d-2296-468c-afb3-30e24c088792)

**Comment by haarse** [comment]
You should run this query in ivorySql:
SELECT o.orderno, c.customername
FROM orders o, customer c
WHERE o.customerid = c.customerid(+);

**Reply by balinorLiang** [reply]
> You should run this query in ivorySql: SELECT o.orderno, c.customername FROM orders o, customer c WHERE o.customerid = c.customerid(+);

Oh, noted. For `(+)` in Oracle, IvorySQL doesn't support now. You could create an issue to for this compatible feature.

**Reply by haarse** [reply]
ok, I have added an enhancement.

Thanks for your work with ivorySql. We will consider migrating if you fix this enhancement.

**Comment by luss** [comment]
I'm a fan of writing sql syntax the older way and therefore am a +1 for adding the outer join systax as an upcoming feature:
SELECT o.orderno, c.customername
FROM orders o, customer c
WHERE o.customerid = c.customerid(+);
